src/intervals/construction.jl

```diff
@@ -103,7 +94,25 @@ struct BareInterval{T<:NumTypes}
         new{T}(_normalisezero(a), _normalisezero(b))
 end
 
+"""
+    _unsafe_bareinterval(T<:NumTypes, lo, hi)
+
+Internal constructor which assumes that `is_valid_interval(lo, hi) == true`.
+
+!!! danger
+    This constructor is **not** compliant with the IEEE Standard 1788-2015.
+    Since misuse of this function can deeply corrupt code, its usage is
+    **strongly discouraged** in favour of [`bareinterval`](@ref).
+"""
+_unsafe_bareinterval
+
 _normalisezero(a) = ifelse(iszero(a), zero(a), a)
+# used only to construct intervals
+_inf(a::BareInterval) = a.lo
+_sup(a::BareInterval) = a.hi
+_inf(a::Real) = a
+_sup(a::Real) = a
```

Comment:
Thanks a lot for renaming these functions! I sort of dislike `inf(a)`, because it makes me think on `Inf`...